app/services/catalog/service_plan_json.rb

```diff
@@ -0,0 +1,37 @@
+module Catalog
+  class ServicePlanJson
+    include JsonSchemaReader
+
+    attr_reader :json
+
+    def initialize(opts)
+      if opts.key?(:service_plan_id)
+        @service_plans = ServicePlan.where(:id => opts[:service_plan_id])
+      elsif opts.key?(:portfolio_item_id)
+        @service_plans = ServicePlan.where(:portfolio_item_id => opts[:portfolio_item_id])
+      end
+
+      @schema = opts[:schema] || "modified"
+      @collection = opts[:collection]
+      @json = []
+    end
+
+    def process
+      @service_plans.each do |plan|
+        @reference = plan.portfolio_item.service_offering_ref
+        @portfolio_item_id = plan.portfolio_item.id
+        @service_plan = OpenStruct.new(
+          :id                 => plan.id,
+          :create_json_schema => plan.send(@schema),
+          :name               => plan.name,
+          :description        => plan.description
+        )
+
+        @json << read_json_schema("service_plan.erb")
+      end
+
+      @json = @json.first unless @collection
```

Comment:
I wonder if the `@json` manipulation ( collection or not ) should live in a method as opposed to declaring an `attr_reader`. The method could better handle the logic around returning an array or not in my opinion.

Reply:
Moved it into a method that returns only the first element unless the `:collection` option is set.